### PR TITLE
API-20687: vfq sends duz to charon

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/RestCharonClient.java [hunk 1]
```diff
@@ -131,10 +135,29 @@ public <I extends TypeSafeRpcRequest, O> CharonResponse<I, O> request(
...
     var details = request.rpcRequest().asDetails();
-    var principal =
-        rpcPrincipalLookup
-            .findByNameAndSite(details.name(), request.vista())
-            .orElseThrow(() -> new ResourceExceptions.NotFound("Site " + request.vista()));
+    RpcPrincipalV1 principal;
+    if (request.maybeDuz().isPresent()) {
+      var principalCodes =
+          rpcPrincipalLookup
+              .findBySiteAndUser(request.vista(), "LHS,APPLICATION PROXY")
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/RestCharonClient.java:142]: Feels like this should be configurable.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/RestCharonClient.java [hunk 2]
```diff
@@ -131,10 +135,29 @@ public <I extends TypeSafeRpcRequest, O> CharonResponse<I, O> request(
...
+              .stream()
+              .map(RpcPrincipalsV1.PrincipalEntry::codes)
+              .flatMap(Collection::stream)
+              .filter(code -> code.sites().contains(request.vista()))
+              .findFirst()
+              .orElseThrow(() -> new ResourceExceptions.NotFound("Site " + request.vista()));
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/RestCharonClient.java:148]: This is kinda misleading, isn't it?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/CharonRequest.java [hunk 3]
```diff
@@ -5,11 +5,21 @@
...
 @SuppressWarnings("ClassCanBeRecord")
 @Value
 @Builder
 public class CharonRequest<I extends TypeSafeRpcRequest, O> {
   @NonNull String vista;
   @NonNull I rpcRequest;
   @NonNull Class<O> responseType;
+  Optional<String> duz;
+
+  Optional<String> maybeDuz() {
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/CharonRequest.java:19]: This is a lazy getter. It should have the same name as the field.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4SiteAppointmentController.java [hunk 4]
```diff
@@ -65,7 +65,9 @@ public Appointment appointmentRead(
             .type(Set.of(VprGetPatientData.Domains.appointments))
             .id(Optional.of(ids.recordId()))
             .build();
-    var charonResponse = charonClient.request(vprGetPatientData(site, rpcRequest));
+    var charonResponse =
+        charonClient.request(
+            vprGetPatientData(site, rpcRequest, Optional.ofNullable(request.getHeader("duz"))));
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4SiteAppointmentController.java:70]: This bit of code to create an optional DUZ is used _a lot_ and the string value is written out every time. That feels like a recipe for major refactor in the future if we decide the header value is different.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/RestCharonClient.java [hunk 5]
```diff
@@ -131,10 +136,33 @@ public <I extends TypeSafeRpcRequest, O> CharonResponse<I, O> request(
...
+              .findBySiteAndUser(request.vista(), LIGHTHOUSE_APPLICATION_PROXY_USER)
+              .stream()
+              .map(RpcPrincipalsV1.PrincipalEntry::codes)
+              .flatMap(Collection::stream)
+              .filter(code -> code.sites().contains(request.vista()))
+              .findFirst()
+              .orElseThrow(
+                  () ->
+                      new ResourceExceptions.NotFound(
+                          "Site " + request.vista() + " for DUZ " + duz.get()));
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/charonclient/RestCharonClient.java:153]: Site and user. The DUZ doesn't factor in here.

**Comment by ilaflamme-va** [comment]
LGTM

**Pull request merged**